sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetConverter.scala

```diff
@@ -426,6 +423,33 @@ private[parquet] class CatalystPrimitiveConverter(
     parent.updateLong(fieldIndex, value)
 }
 
+/**
+ * A `parquet.io.api.PrimitiveConverter` that converts Parquet Binary to Catalyst String.
+ * Supports dictionaries to reduce Binary to String conversion overhead.
+ *
+ * Follows pattern in Parquet of using dictionaries, where supported, for String conversion.
+ *
+ * @param parent The parent group converter.
+ * @param fieldIndex The index inside the record.
+ */
+private[parquet] class CatalystPrimitiveStringConverter(parent: CatalystConverter, fieldIndex: Int)
+  extends CatalystPrimitiveConverter(parent, fieldIndex) {
+
+  private var dict: Array[String] = null
```

Comment:
I have a hard time believing this really matters (eg., if its actually hot, then the JIT will take care of it), but if we do care about this, should we recommend that people *always* use `private[this]`?? I have a hard time imaginging that anyone will ever need `private` instead.